ciao-controller/instance.go

```diff
@@ -46,7 +46,7 @@ type instance struct {
 	startTime time.Time
 }
 
-func isCNCIWorkload(workload *types.Workload) bool {
+func isCNCIWorkload(workload types.Workload) bool {
```

Comment:
> I did have another reason as well - and that was that I was trying to reduce the likelyhood that a receiver would modify the struct. It felt safer to send a copy.

Unfortunately, this is really hard to achieve in go even when passing by value, unless the parameter is of a primitive type or is an array or struct containing only simply types.  The reason is the contents of maps and slices are modifiable by a function even if they are passed to that function by value, either directly or part of a struct which is itself passed by value.  For example, it's still possible to modify the caller's copy of the workload parameter inside this function even though it is passed by value

workload.Default[0].Type = payloads.VCPUs

This change will be visible outside isCNCIWorkload.

You could always change the parameter of this function to take a []payloads.RequestedResource.  That wouldn't prevent the function from modifying the slice but would bypass the value vs pointer discussion.